pkg/comp-functions/functions/common/resources.go

```diff
@@ -3,43 +3,111 @@ package common
 import (
 	vshnv1 "github.com/vshn/appcat/v4/apis/vshn/v1"
 	"github.com/vshn/appcat/v4/pkg/common/utils"
+	"k8s.io/apimachinery/pkg/api/resource"
 )
 
 type Resources struct {
-	ReqMem string
-	ReqCPU string
-	Mem    string
-	CPU    string
-	Disk   string
+	ReqMem resource.Quantity
+	ReqCPU resource.Quantity
+	Mem    resource.Quantity
+	CPU    resource.Quantity
+	Disk   resource.Quantity
 }
 
-func GetResources(size *vshnv1.VSHNSizeSpec, r utils.Resources) Resources {
-	reqMem := size.Requests.Memory
-	reqCPU := size.Requests.CPU
-	mem := size.Memory
-	cpu := size.CPU
-	disk := size.Disk
+func GetResources(size *vshnv1.VSHNSizeSpec, plan utils.Resources) (Resources, error) {
+	reqMem := resource.Quantity{}
+	reqCPU := resource.Quantity{}
+	mem := resource.Quantity{}
+	cpu := resource.Quantity{}
+	disk := plan.Disk
```

Comment:
Don't we have the same problem with the disk as well? For disk we should always get the highest value, we cannot go back on a lower value

Reply:
No. Because we need to be able to set a lower disk size than the plan provides on initial provisioning.
I would ignore the "lowering" disk space problem for now. It won't break anything. It will just not do anything.
Afaik documenting that lowering disk space is not possible should be enough.

Reply:
I guess it's fine by me.

Reply:
can You cover that case in e2e? so we're sure that change doesn't affect end user with time (in case Crossplane introduce feature that will break things in that field)